app/controllers/application_controller.rb

```diff
@@ -2,7 +2,7 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
   def current_user
-    @current_user ||= User.find(session[:user_id])
+    @current_user ||= session[:user_id].present? ? User.find(session[:user_id]) : nil
```

Comment:
`@current_user ||= User.find(session[:user_id]) if session[:user_id].present?` nets the same result, but I'll understand if you prefer the explicit `nil` return here.